system_zero/src/alu/div-constraints-proof.txt

```diff
@@ -0,0 +1,94 @@
+Constraints A:
+A1. 0 <= dividend <= u32::MAX
```

Comment:
Just curious why the `0 <=`s? I think it's implied that these variables are in the field `Z/pZ`, in which case negatives just don't exist right? Though doesn't hurt if you prefer it.

Reply:
Oh I see, steps like `then divisor - remainder - 1 < 0` (in B7) are written in terms of integer arithmetic, not field arithmetic. I think to be fully rigorous we would want the latter, which would slightly complicate steps like that (since it holds for sufficiently large fields). Maybe it's alright to skip those details though.

Reply:
Hmm, maybe this works as a fully rigorous proof if we take our field representatives to be {-(p-1)/2, ..., -1, 0, 1, ..., (p-1)/2} rather than the usual {0, 1, ..., p-1}? Then lifting field elements to integers and comparing them makes sense with negatives.